eventio/event_io_file.py

```diff
@@ -0,0 +1,65 @@
+import struct
+from .object_header import make_ObjectHeader
+from io import BytesIO
+
+class EventIOObject:
+    ''' A generic EventIOObject
+
+    It has a list of `headers` and a binary string `payload`.
+    The payload might be loaded lazily on first access or already in memory.
+    This is decided on construction time.
+    '''
+    def __init__(self, headers, payload=None, file=None):
```

Comment:
the classmethod approach as discussed here:
http://stackoverflow.com/q/682504/1532474
has zero upvotes and the author says: "if the constructors do not share code", but we can think about it.

However, I am currently thinking there is reason to always give it the file, and let it load the contents itself. If the user goes linearly through the file, the file pointer is already at the right position, so the `self._file.seek` is basically a no-op, just the calculation of the start_adress costs a bit, since it is not needed, when the pointer is already at the right position. (Who cares?!)

I will try that now ...